src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp

```diff
@@ -174,6 +179,10 @@ namespace OpenMS
       {
         spec_c = true;
       }
+      else if (allow_random_asp_pro_cleavage && (sequence[end-1] == 'D') && (sequence[end] == 'P'))
```

Comment:
can end-1 < 0?

Reply:
no, see line 139. length >=1 and end = pos + length